h2/src/main/org/h2/command/Parser.java

```diff
@@ -5956,6 +5957,17 @@ private DefineCommand parseAlterTableAddConstraintIf(String tableName,
                 read("BTREE");
             }
             return command;
+        } else if (allowAffinityKey && readIf("AFFINITY")) {
+            read("KEY");
+            IndexColumn[] indexColumns = new IndexColumn[] { new IndexColumn() };
+            indexColumns[0].columnName = readColumnIdentifier();
```

Comment:
I do not like that we have a separate logic from PRIMARY KEY branch and for some reason support only a single column here. Lets reuse the code for PRIMARY KEY parsing.